### PR TITLE
docs(schema): correct example for `Schema.prototype.discriminator()`

### LINKED ISSUE #12455: `Schema.prototype.discriminator` example is not showing usage of correct function
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the issue has not already been raised

### Issue

Currently the example in the JSDOC for `Schema.prototype.discriminator` does not actually show usage of the `Schema.prototype.discriminator` usage, but usage of `Model.discriminator`, see

https://github.com/Automattic/mongoose/blob/4495cf3a407a1fd6443b28a437800ed6a1b3c133/lib/schema.js#L499-L532

### PR DESCRIPTION
**Pull request by vkarpov15**
Fix #12455

**Summary**

Quick docs fix since the current example doesn't use the actual function the docs are describing

**Examples**

### CODE REVIEW

File: lib/schema.js [hunk 1]
```diff
@@ -502,25 +502,20 @@ Schema.prototype.defaultOptions = function(options) {
...
+ *     Event.discriminators['clicked']; // Model { clicked }
  *
- *     // When you create a generic event, it can't have a URL field...
- *     const genericEvent = new Event({ time: Date.now(), url: 'google.com' });
- *     assert.ok(!genericEvent.url);
- *     // But a ClickedLinkEvent can
- *     const clickedEvent = new ClickedLinkEvent({ time: Date.now(), url: 'google.com' });
- *     assert.ok(clickedEvent.url);
+ *     const doc = await Event.create({ kind: 'clicked', element: '#hero' });
+ *     doc.element; // '#hero'
```
Comment (hasezoey) [inline lib/schema.js:515]:
```suggestion
 *     doc.element; // '#hero'
 *     doc instanceof ClickedModel; // true
```

File: lib/schema.js [hunk 2]
```diff
@@ -502,25 +502,20 @@ Schema.prototype.defaultOptions = function(options) {
  *
  * #### Example:
  *
- *     const options = { discriminatorKey: 'kind' };
+ *     const eventSchema = new mongoose.Schema({ timestamp: Date }, { discriminatorKey: 'kind' });
+ *
+ *     const clickedEventSchema = new mongoose.Schema({ element: String }, { discriminatorKey: 'kind' });
+ *     eventSchema.discriminator('clicked', clickedEventSchema);
```
Comment (hasezoey) [inline lib/schema.js:508]:
```suggestion
 *     const ClickedModel = eventSchema.discriminator('clicked', clickedEventSchema);
```

**Review by hasezoey — APPROVED** [review]
LGTM

maybe add the following suggestions to make it clear that the document is a document of the discriminated model

**Review by AbdelrahmanHafez — APPROVED** [review]
LGTM, thanks 👍

**Pull request merged**